File: src/main/java/fr/spoutnik87/musicbot_rest/security/UserDetails.java
package fr.spoutnik87.musicbot_rest.security;

import fr.spoutnik87.musicbot_rest.model.User;
import lombok.Data;
import org.springframework.security.core.GrantedAuthority;

import java.util.Collection;

@Data
public class UserDetails extends org.springframework.security.core.userdetails.User {
    private User user;

    public UserDetails(User user, Collection<? extends GrantedAuthority> grantedAuthorities) {
        super(user.getEmail(), user.getPassword(), grantedAuthorities);
        this.user = user;
    }
}

File: src/main/kotlin/fr/spoutnik87/musicbot_rest/security/UserDetails.kt
package fr.spoutnik87.musicbot_rest.security

import fr.spoutnik87.musicbot_rest.model.User
import org.springframework.security.core.GrantedAuthority

data class UserDetails(
        var user: User,
        var grantedAuthorities: Collection<GrantedAuthority>
) : org.springframework.security.core.userdetails.User(user.email, user.password, grantedAuthorities) {
}